### ISSUE TITLE
How to get txt files of the predictions on my test set

### ISSUE THREAD

**Issue by KAKAROT12419** [ultralytics/ultralytics#10468]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

I have trained yolov8 models on training set. Now i want to predict and get the txt files of prediction. Can anybody help me please  and tell me how to get the txt files of the prediction.

**Comment by glenn-jocher** [comment]
Hello! 😊

To save your YOLOv8 predictions as `.txt` files on your test set, you'll want to use the predict mode with the `save_txt=True` argument. Here's how you can do it:

### Using Command Line Interface (CLI):

```bash
yolo predict model=path/to/your/model.pt source=path/to/your/test/images save_txt=True
```

### Using Python:

```python
from ultralytics import YOLO

# Load your trained model
model = YOLO('path/to/your/model.pt')

# Make predictions and save results as .txt
model.predict(source='path/to/your/test/images', save_txt=True)
```

This will generate `.txt` files for each image in your test set, where each file contains the detections in the format `[class] [confidence] [x_center] [y_center] [width] [height]`, all normalized to the image width and height.

Hope this helps! Let us know if you have any further questions. 🌟

**Comment by KAKAROT12419** [comment]
Thankyou sir, but when i am storing confidence score txt file is giving [class] [x_center] [y_center] [width] [height] [confidence]. but this is not the output format of yolov8 why is it coming so.

**Comment by glenn-jocher** [comment]
Hello! 😊 Glad to see you're making progress with your YOLOv8 model predictions. The format `[class] [x_center] [y_center] [width] [height] [confidence]` that you're observing in the .txt files is actually a standard output format provided by the YOLOv8 framework when using the `save_txt=True` option. This format is designed to offer a detailed insight into the prediction by including both the bounding box coordinates and the confidence score for each detected object.

If you're expecting a different format or have specific requirements, could you please share more details? We’re here to help!

**Comment by amarnathreddy0201** [comment]
detecting object
yolo detect predict model=yolov8x.pt source='video file' save=True save_txt=True classes=0

I got data values in Txt file.
0 0.350915 0.342867 0.0668488 0.251985
0 0.491775 0.314995 0.0615532 0.220196
0 0.634487 0.348869 0.0608536 0.0987466

**Comment by glenn-jocher** [comment]
The issue you're encountering is likely due to the `save_conf` parameter, which controls whether confidence scores are included in the output. By default, `save_conf` is set to `False`. To include confidence values in your `.txt` files, modify your command like this:

```bash
yolo detect predict model=yolov8x.pt source='video file' save=True save_txt=True classes=0 save_conf=True
```

This should include the confidence scores in the output format `[class] [x_center] [y_center] [width] [height] [confidence]`. Let me know if you need further assistance!
